### DISCUSSION TITLE
Userinfo endpoint not giving infos for outline and portainer

### DISCUSSION THREAD

**Discussion by FRFlo** [ory/hydra#3799] [Q&A] [answered]
Hi dear community,

Today i'm moving from internal to OIDC auth on my [Portainer](https://www.portainer.io/) instance and on my [Outline](https://www.getoutline.com/) instance. But i have difficulties on this.

It seems like both of my apps cannot retrieve user's information (traits like "email" or "preferred_username"). Outline is for sure wanting this type of response :
```JSON
{
  "email": "string",
  "preferred_username": "string"
}
```
but it is receiving this instead:
```JSON
{
  "amr": [
    "password"
  ],
  "aud": [
    "string"
  ],
  "auth_time": "number",
  "iat": "number",,
  "iss": "string",
  "rat": "number",,
  "sub": "string"
}
```

Portainer isn't showing the username of the user who failed to login so i assume it's the same issue.

Am i going to the good endpoint ? (https://hydra.example.com/userinfo)

**Comment by terev** [comment, marked as answer]
My guess is the `email` and `profile` scopes aren't being granted for some reason. The reason for this may include:
- Your OAuth client isn't configured to allow requesting the `email` and `profile` scopes.
- The user agent isn't requesting these scopes.
- The user isn't consenting to these scopes.
